drivers/linstorvhdutil.py

```diff
@@ -120,14 +111,45 @@ def wrapper(*args, **kwargs):
             remote_args = {str(key): str(value) for key, value in remote_args.items()}
 
             try:
-                def remote_call():
-                    host_ref = self._get_readonly_host(vdi_uuid, device_path, node_names)
-                    return call_remote_method(self._session, host_ref, remote_method, device_path, remote_args)
-                response = util.retry(remote_call, 5, 2)
-            except Exception as remote_e:
-                self._raise_openers_exception(device_path, local_e or remote_e)
+                host_ref_attached = list(util.get_hosts_attached_on(self._session, [vdi_uuid]))[0]
```

Comment:
```suggestion
                host_ref_attached = next(iter(util.get_hosts_attached_on(self._session, [vdi_uuid])))
```

Reply:
fixed in [365ae23](https://github.com/xcp-ng/sm/pull/93/commits/365ae231feaa4ea01fa386ba0f43f4499531d1b5)